cupy/_core/core.pyx

```diff
@@ -60,6 +60,20 @@ from cupy.exceptions import ComplexWarning
 NUMPY_1x = numpy.__version__ < '2'
 
 
+cdef extern from *:
+    """
+    #define _str_(s) #s
+    #define _xstr_(s) _str_(s)
+    const char* cupy_cache_key = _xstr_(CUPY_CACHE_KEY);
+    #undef _xstr_
+    #undef _str_
+    """
+    const char* cupy_cache_key  # set at build time
+
+
+CUPY_CACHE_KEY = bytes(cupy_cache_key).decode()
```

Comment:
Cython knows how to decode `const char*`. So writing this without the `bytes` copy works

Also we can type this as `str`, which can help Cython generate more specific CPython code 

```suggestion
cdef str CUPY_CACHE_KEY = cupy_cache_key.decode()
```